### PR TITLE
Add batch and non-SetToken 0x TradeQuote fetching 

### PR DESCRIPTION
**Pull request by cgewecke**
PR adds:
+ `fetchSwapQuoteAsync` : a method to fetch simple zero ex swap quote for any token pair
+ `batchFetchSwapQuotesAsync`: batch request version of `fetchSwapQuote...`

**TODO**
- [x] Unit tests
- [x] Extract methods from Trade into a new "utils" or "periphery" API (that includes coingecko and gas oracle methods) 
- [x] If re-org'd as proposed, this should be published as new major version

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "set.js",
-    "version": "0.4.12",
+    "version": "0.4.12-batch.16",
```
Comment (asoong) [inline package.json:3]: Update this before merging?

File: src/api/UtilsAPI.ts [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+      excludedSources,
+    });
+  }
+
+  /**
+   * Call 0x API to generate a trade quote for two SetToken components.
+   *
+   * 0x rate-limits calls per API key as follows:
+   *
+   * > Ethereum: 10 requests per second/200 requests per minute.
```
Comment (asoong) [inline src/api/UtilsAPI.ts:134]: Update this?

**Review by asoong — APPROVED** [review]
LGTM

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "set.js",
-    "version": "0.4.12",
+    "version": "0.4.12-batch.14",
```
[the remark this answers is not in the document]
Comment (cgewecke) [reply package.json]: Yes this is just so I can use this branch at set-ui (it's a pre-release tag)

**Pull request merged**